File: android/src/main/kotlin/lazebny/io/jc/common/JcViewController.kt
package lazebny.io.jc.common

import io.flutter.plugin.common.BinaryMessenger
import io.flutter.plugin.common.MethodChannel

interface JcViewController {
    fun setLayoutParams(width: Double, height: Double)

    companion object {
        fun setUp(
            binaryMessenger: BinaryMessenger,
            viedId: Int,
            viewController: JcViewController?
        ) {
            val methodChannel = MethodChannel(
                binaryMessenger,
                "lazebny.io.jc/JcViewController/$viedId",
            )

            if (viewController == null) {
                methodChannel.setMethodCallHandler(null)
                return
            }

            methodChannel.setMethodCallHandler { call, result ->
                when (call.method) {
                    "setLayoutParams" -> {
                        val width = call.argument<Double>("width")
                        val height = call.argument<Double>("height")
                        viewController.setLayoutParams(width!!, height!!)
                        result.success(null)
                    }

                    else -> {
                        result.notImplemented()
                    }
                }
            }
        }
    }
}

File: ios/Classes/common/JcViewController.swift
//
//  JcViewController.swift
//  jc
//
//  Created by Михаил on 30.05.2023.
//

import Foundation
import Flutter

protocol JcViewController {
    func setLayoutParams(width: Double, height: Double)
}

class JcViewControllerSetup {
    static func setUp(binaryMessenger: FlutterBinaryMessenger, viewId: Int64, viewController: JcViewController?) {
        let methodChannel = FlutterMethodChannel(name: "lazebny.io.jc/JcViewController/\(viewId)",
                                                 binaryMessenger: binaryMessenger)

        if viewController == nil {
            methodChannel.setMethodCallHandler(nil)
            return
        }

        methodChannel.setMethodCallHandler { (call: FlutterMethodCall, result: @escaping FlutterResult) -> Void in
            switch call.method {
            case "setLayoutParams":
                if let args = call.arguments as? [String: Double],
                   let width = args["width"],
                   let height = args["height"] {
                    viewController?.setLayoutParams(width: width, height: height)
                    result(nil)
                } else {
                    result(FlutterError(code: "Invalid args", message: "Width or height was not a Double", details: nil))
                }
            default:
                result(FlutterMethodNotImplemented)
            }
        }
    }
}
